cni/pkg/ambient/server.go

```diff
@@ -50,6 +50,9 @@ type Server struct {
 	iptablesCommand lazy.Lazy[string]
 	redirectMode    RedirectMode
 	ebpfServer      *ebpf.RedirectServer
+
+	// podReconcileHandler is implemented by Server, but it can be overridden by tests.
+	podReconcileHandler podReconcileHandler
```

Comment:
maybe i missed it, can you point where do you set it

Reply:
Missed, I think this is why the test failed. Just checked it